lib/active_public_resources/oauth/vimeo.rb

```diff
@@ -0,0 +1,36 @@
+require 'httparty'
+require 'base64'
+
+module ActivePublicResources
+  module OAuth
+    class Vimeo
+      attr_reader :token
+
+      AUTHORIZE_URL = 'https://api.vimeo.com/oauth/authorize/client'
+      VERIFY_URL = 'https://api.vimeo.com/oauth/verify'
+
+      def initialize(consumer_key, consumer_secret)
+        @token = Base64.urlsafe_encode64("#{consumer_key}:#{consumer_secret}")
+      end
+
+      def get_access_token
+        response = HTTParty.post(
+          AUTHORIZE_URL,
+          body: { grant_type: 'client_credentials' },
+          headers: { "Authorization" => "Basic #{@token}" }
+        )
+        return response['access_token']
```

Comment:
Just to alleviate my paranoia It might also be good to explicitly return nil here if the response does not have a 200 response code. We don't want any infinite recursion if `response['acces_token']` is somehow a truthy value when the request fails. 😄